tdms/src/interpolate_Hfield.cpp

```diff
@@ -1,161 +1,231 @@
-        {
-            // using a BLi scheme in z, assign memory we need
-            other_dim_cell_data = (double *)malloc(8 * sizeof(double));
+}
 
-            // Each time we interpolate in y, we need data for cells j - (scheme+1) through j - (scheme+1) + 7, inclusive
-            d_low = j - (z_scheme + 1);
-            d_high = j - (z_scheme + 1) + 7;
-            for (int ii = i_low; ii <= i_high; ii++)
+void interpolateTimeDomainHy(double ***Hyx, double ***Hyz, int i, int j, int k, int nI, int nK, double *Hy)
+{
+    // Associations: a = y, b = z, c = x
+
+    // determine the x-direction scheme
+    const interpScheme &x_scheme = best_interp_scheme(nI, i);
+    // determine the z-direction scheme
+    const interpScheme &z_scheme = best_interp_scheme(nK, k);
+
+    // this data will be passed to the second interpolation scheme
+    double data_for_second_scheme[8];
+    // this data will hold values for the interpolation in the first interpolation scheme
+    double data_for_first_scheme[8];
+
+    // which of x_scheme and z_scheme is WORSE? We will interpolate in this direction SECOND
+    if (z_scheme.is_better_than(x_scheme))
+    {
+        // we will be interpolating in the z-direction first, then in x
+        for (int ii = x_scheme.first_nonzero_coeff; ii <= x_scheme.last_nonzero_coeff; ii++)
+        {
+            // this is the i-index of the cell we are looking at
+            int cell_i = i - (x_scheme.index + 1) + ii;
+            // determine the Hy values of cells (cell_i, j, k-z_scheme.index-1) through (cell_i, j, k-z_scheme.index-1+7), and interpolate them via z_scheme
+            for (int kk = z_scheme.first_nonzero_coeff; kk <= z_scheme.last_nonzero_coeff; kk++)
             {
-                // populate other_dim_cell_data with the data we need from the y-cells
-                for (int kk = d_low; kk <= d_high; kk++)
-                {
-                    other_dim_cell_data[kk - d_low] = Hxy[kk][j][ii] + Hxz[kk][j][ii];
-                }
-                first_interp_vals[ii - i_low] = bandlimited_interpolation(z_scheme, other_dim_cell_data);
+                // the k-index of the current cell we are looking at (readability, don't need to define this here)
+                int cell_k = k - (z_scheme.index + 1) + kk;
+                // gather the data for interpolating in the z dimension
+                data_for_first_scheme[kk] = Hyx[cell_k][j][cell_i] + Hyz[cell_k][j][cell_i];
             }
+            // interpolate in z to obtain a value for the Hy field at position (cell_i+Dx, j, k)
+            // place this into the appropriate index in the data being passed to the x_scheme
+            data_for_second_scheme[ii] = z_scheme.interpolate(data_for_first_scheme);
         }
-        else
+        // now interpolate in the x-direction to the centre of Yee cell (i,j,k)
+        *Hy = x_scheme.interpolate(data_for_second_scheme);
+    }
+    else
+    {
```

Comment:
Marginal and/or nit-picky comment, but [Google style](https://google.github.io/styleguide/cppguide.html#Conditionals) prefers...
```c++
if (condition) {                   // no spaces inside parentheses, space before brace
  DoOneThing();                    // two space indent
  DoAnotherThing();
} else if (int a = f(); a != 3) {  // closing brace on new line, else on same line
  DoAThirdThing(a);
} else {
  DoNothing();
}
```
what editor do you use? Can you try `clang-format` (or similar). 

In case of vim there's a [Google C++ style file](https://github.com/vim-scripts/google.vim) you can use. In case of VSCode it should happen by default(?). In case of emacs, you're on your own 😛

Reply:
Obviously apply style rules iff appropriate for clarity etc.

Reply:
This is the result of VS Code's hideous default formatting that I have now overwritten :sweat_smile:

The style I've been _writing_ is:
```c++
if (condition) {       // no spaces inside parentheses, space before brace 
    DoOneThing();      // 4 space indent 
    DoAnotherThing(); 
} 
else if (int a = f(); a != 3) {  // closing brace on new line, else on same line 
    DoAThirdThing(a); 
} 
else {
    DoNothing(); 
}
```
however what you're seeing here is the result of that being run through VSCode's formatting after editing a seemingly unrelated line of code...